src/Interpreters/Session.h

```diff
@@ -113,6 +113,7 @@ class Session
 
     mutable UserPtr user;
     std::optional<UUID> user_id;
+    AuthenticationData user_authenticated_with;
```

Comment:
I am not quite familiar with the `Session` object lifetime, but this field is filled upon authentication and is only used in `recordLoginSuccess`, so I assume it is safe to use it as is.